src/Curry.php

```diff
@@ -0,0 +1,77 @@
+<?php
+declare(strict_types=1);
```

Comment:
It [does more than that](https://secure.php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration.strict), including forcing return type checks.